lib/connection.ts

```diff
@@ -227,6 +227,7 @@ export class Connection extends EventEmitter {
 
         const removeListeners: Function = () => {
           clearTimeout(waitTimer);
+          this.actionInitiated = false;
```

Comment:
is it possible for these actions to overlap? i wouldn't store these on the class because if two actions are running concurrently, they will clash over this flag

Reply:
if you just want to know whether _any_ action is running, you could keep an `actionCount` instead, incrementing it at the beginning of an action and decrementing it at the end. if it's 0, then there's nothing going on

Reply:
That's true. Would it be better for this to be a counter instead of a flag?